cpp/tests/strings/json_tests.cpp

```diff
@@ -759,3 +759,109 @@ TEST_F(JsonTests, MixedOutput)
     CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(*result, expected);
   }
 }
+
+TEST_F(JsonTests, StripQuotes)
+{
+  // we normally expect our outputs here to be
+  // b     (no quotes)
+  // but with string_quotes_from_single_strings false, we expect
+  // "b"   (with quotes)
+  {
+    std::string str("{\"a\" : \"b\"}");
+    cudf::test::strings_column_wrapper input({str, str});
+
+    cudf::strings::get_json_object_options options;
+    options.set_strip_quotes_from_single_strings(false);
+
+    std::string json_path("$.a");
+    auto result_raw =
+      cudf::strings::get_json_object(cudf::strings_column_view(input), json_path, options);
+    auto result = drop_whitespace(*result_raw);
+
+    cudf::test::strings_column_wrapper expected_raw({"\"b\"", "\"b\""});
+    auto expected = drop_whitespace(expected_raw);
+
+    CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(*result, *expected);
+  }
+
+  // a valid, but empty row
+  {
+    cudf::test::strings_column_wrapper input{"{\"store\": { \"bicycle\" : \"\" } }"};
+    std::string json_path("$.store.bicycle");
+
+    cudf::strings::get_json_object_options options;
+    options.set_strip_quotes_from_single_strings(true);
+
+    auto result =
+      cudf::strings::get_json_object(cudf::strings_column_view(input), json_path, options);
+
+    cudf::test::strings_column_wrapper expected({""});
+
+    CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(*result, expected);
+  }
+}
+
+TEST_F(JsonTests, AllowSingleQuotes)
+{
+  // Tests allowing single quotes for strings.
+  // Note:  this flag allows a mix of single and double quotes. it doesn't explicitly require
+  // single-quotes only.
+
+  // various queries on:
+  // clang-format off
+  std::vector<std::string> input_strings {
+    "{\'a\': {\'b\' : \'c\'}}",
+
```

Comment:
Perhaps include a test string like ` "abc'def" `.
Otherwise, a simple global replace of `'` with `"` before calling `get_json_object` would work for this.